### DISCUSSION TITLE
why does command `pnpm config set registry` modifies the user npmrc file?

### DISCUSSION THREAD

**Discussion by mzleman** [pnpm/pnpm#8893] [Q&A]
I found a comfusing behaviour of pnpm.
**description：**
 1. my os is windows
 2. my pnpm version is v9.12.1
 3. my npm user  config file path is ~/.npmrc
 4. and apparently my pnpm global config file path is ~/AppData/Local/pnpm/config/rc
 
 when i ran command `pnpm config --global set proxy  ...`, it modified pnpm global config file.
 but when i ran command `pnpm config --global set registry ...`, it modified npm user config file.
 
 **questions：**
Was it designed that way？
My expectation is that when I use the `pnpm config --global set ` command, it should always modify the pnpm global file.
Is "registry" the only field that is special?

**Comment by zkochan** [comment]
I am not sure why it happens. I think I would expect both to modify the global pnpm file.

**Reply by mzleman** [reply]
thanks for comment. I think these codes make it happen. The `--location` parameter value in npm is  "user" by default.
![PixPin_2024-12-24_09-21-33](https://github.com/user-attachments/assets/e5a3d7df-ffd5-4e7c-916f-91f139e7a974)
